debian/ubuntu-advantage-tools.postinst

```diff
@@ -265,6 +265,24 @@ mark_reboot_cmds_as_needed() {
     add_notice "$msg_name"
 }
 
+check_wrong_fips_metapackage_on_cloud() {
+    fips_metapkg="ubuntu-fips"
+
+    cloud_id=""
+    if command -v "cloud-id" > /dev/null ; then
+      cloud_id=$(cloud-id)
+    fi
+
+    # If the package is not installed, we don't want the postinst script to fail
+    fips_installed=$(dpkg-query -W --showformat='${db:Status-Status}\n' $fips_metapkg 2>/dev/null || true)
+
+    if echo "$cloud_id" | grep -E -q "^(azure|aws)"; then
+      if echo "$fips_installed" | grep -E -q "installed"; then
```

Comment:
@lucasmoura Don't we also need to check too if we are on bionic or greater? Xenial doesn't have a cloud-optimized kernel on most clouds.

Reply:
Good catch, we do have a way to unblock installing fips on xenial. Because of that, we need that bionic >= check
I will update the code